src/Collapse.js

```diff
@@ -42,10 +42,14 @@ const Collapse = React.createClass({
 
 
   onHeightReady(height) {
-    if (this.renderStatic && this.props.isOpened) {
-      this.height = stringHeight(height);
+    const isOpened = this.props.isOpened;
+    let _height = height;
+    if (this.renderStatic && isOpened) {
+      this.height = stringHeight(_height);
+    } else if (!isOpened) {
+      _height = 0;
     }
-    this.setState({height});
+    this.setState({_height});
```

Comment:
My fault: i'm editing directly in browsery and fix code from github editor. Too lazy to deploy :)